### ISSUE TITLE
Use /members/:handle to save basic_info traits that are associated with member profile

### ISSUE THREAD

**Issue by rakibansary** [topcoder-platform/micro-frontends-onboarding-app#166]
The following attributes are saved in DynamoDB table: MemberProfile

- addresses
- competitionCountryCode
- homeCountryCode

Onboarding app should be updated to use v5/member/:handle Api `addresses` attribute to store data entered on the page https://platform.topcoder-dev.com/onboard/contact-details

**Comment by SathyaJayabal** [comment]
@rakibansary , connect data, timezone and working hours are not getting saved

**Comment by SathyaJayabal** [comment]
also the skills PATCH endpoint is giving this error
<img width="1440" alt="Screenshot 2021-12-22 at 11 35 47 AM" src="https://user-images.githubusercontent.com/58783823/147044170-47dc9225-2fed-448e-8229-c240aa863e60.png">

**Comment by sandhiyakavi** [comment]
Adding to the above,

Validation messages are appearing for all the Address related fields (mentioning it as mandatory fields) and user is not able to proceed to Step 3 without entering those details.

![image (18)](https://user-images.githubusercontent.com/55895538/147045146-706b600a-b28c-4fcd-b304-14573df4e226.png)

**Comment by SathyaJayabal** [comment]
@rakibansary , there is something wrong with the way the profile fields are saved. I am getting an error when trying to get member profile for a user for whom we have saved using the onbaording flow
example: WM
<img width="1440" alt="Screenshot 2021-12-22 at 5 51 24 PM" src="https://user-images.githubusercontent.com/58783823/147093866-a7c34bad-106d-4d18-9b79-ed56bcd516ad.png">

**Comment by Oanh-and-only-Oanh** [comment]
@rakibansary @SathyaJayabal,
1)none of the wizard fields should be mandatory and user should be able to progress without completing it.
2 Connect data, timezone and working hours should get saved

**Comment by vikasrohit** [comment]
what is connect data @SathyaJayabal ?

**Comment by SathyaJayabal** [comment]
@vikasrohit , it is the timezone and working hours we were saving in connect_info traits, it is not getting saved now.

**Comment by rakibansary** [comment]
@Oanh-and-only-Oanh , @vikasrohit 

1. No field is now mandatory
2. Save issues have been resolved
3. v5 member api did not have a way to add skills for newly created users. v5 member Api had a PATCH endpoint that could only be used to add additional skills. A new endpoint to allow for adding a first skill has been added to v5 member Api in this PR: https://github.com/topcoder-platform/member-api/pull/71

**Comment by sandhiyakavi** [comment]
@rakibansary Below issues are found

1. This issue exists https://github.com/topcoder-platform/micro-frontends-onboarding-app/issues/166#issuecomment-999546205 . 
2. The Community app settings page is not displaying (showing error page) once the address details are saved in Onboarding app.

![image](https://user-images.githubusercontent.com/55895538/147901562-6e0634b8-c51d-4ab1-80c2-7f55b21f79d9.png)

3. The added skills removed from the main page are not getting removed .
    Note: the skills if removed from the Add skills module are getting removed.


https://user-images.githubusercontent.com/55895538/147901751-25aeef8a-4eca-42c3-a213-cd229ca1f5db.mp4

4. Country field changed in Step 2 is not getting displayed in the Connect app now.

![image](https://user-images.githubusercontent.com/55895538/147901816-543cd274-3e4a-4f77-8eec-5afbfa97fa6a.png)

![image](https://user-images.githubusercontent.com/55895538/147901828-9b1bf890-d8ea-419a-b861-0975d8f7a475.png)

5. Primary interest entered in Onboarding app is not showing in the Community app now.

**Comment by SathyaJayabal** [comment]
6. one more @rakibansary , skills added in Onboarding not showing up in settings page

**Comment by rakibansary** [comment]
Thanks @sandhiyakavi / @SathyaJayabal 

1. This seems to be specific to v3 Api. We have three potential solutions. a) Patch v3 Api itself, b) Update all places that are still using v3 like the Work Manager c) Update legacy processor to not send data to v3 Api that it wasn't originally coded handle - I'm analyzing pros/cons of (a) vs (c)
2. Fixed
3. Fixed
4. This is intentional. Connect app fetches Country from users `connect_info` trait. Onboarding app only writes Time Zone, Start Time & End Time to the `connect_info` trait.
5. Fixed
6. Fixed

**Comment by sandhiyakavi** [comment]
@rakibansary For issue No:5 (Primary interest entered in Onboarding app is not showing in the Community app now.)
It is currently not showing in Community app while entering the first time in Onboarding app, showing only after we edit the first entered primary interests details.


https://user-images.githubusercontent.com/55895538/147918883-a6bde848-4cab-48ab-b33d-c5e622ab40f0.mp4

2. If any address related field entered is removed in Onboarding app, it is displaying as undefined in Community app.
<img width="960" alt="undefined_1" src="https://user-images.githubusercontent.com/55895538/147919559-382908cb-3065-493a-8505-ba8b36bfb02d.png">
<img width="960" alt="undefined_2" src="https://user-images.githubusercontent.com/55895538/147919565-bd0af773-e54d-4e31-a1ca-58a97be2e50d.png">

3. Another issue related to short bio, it is not getting displayed in the Community app.
![image](https://user-images.githubusercontent.com/55895538/147919730-e1bd4076-811e-4ec0-bd8e-7d7803ac83de.png)
![image](https://user-images.githubusercontent.com/55895538/147919771-9d473a59-d9fe-47af-b8f9-58045b9703fa.png)


The other issues 2, 3, 6 are working now.

**Comment by rakibansary** [comment]
@sandhiyakavi 

1 & 2 have been fixed in [this commit](https://github.com/topcoder-platform/community-app/commit/c43e31b9128c6b48126f83ddfba4923922271eaf)

3 has been fixed in [this commit](https://github.com/topcoder-platform/micro-frontends-onboarding-app/commit/d06feb02f5d9ed89b7f13dc3f7d907b2b870c6f9). bio is now saved in profile object (/v5/members/:handle).

**Comment by sandhiyakavi** [comment]
Verified in Dev Env. All issues mentioned in https://github.com/topcoder-platform/micro-frontends-onboarding-app/issues/166#issuecomment-1003993221 are resolved.

Only this issue https://github.com/topcoder-platform/micro-frontends-onboarding-app/issues/166#issuecomment-999546205 still exists

**Comment by sandhiyakavi** [comment]
Passing this issue as the issue https://github.com/topcoder-platform/micro-frontends-onboarding-app/issues/166#issuecomment-999546205 is logged separately https://github.com/topcoder-platform/micro-frontends-onboarding-app/issues/175

**Comment by sandhiyakavi** [comment]
Verified in Prod Env. Working as expected.
